### PR TITLE
feat: ad placeholder block

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Implements the "Newsletter Ad" block for manual selection and placement of ads in the newsletter:

image

image

For the "automatic selection", the selection is based on the position percentage. The ads are sorted by position and, regardless of the position the block is placed, the first positioned block will render the first position percentage.

### How to test the changes in this Pull Request:

1. Check out this branch and make sure you have active newsletter ads (at least 2)
2. Draft a new newsletter and place a "Newsletter Ad" block
3. Confirm it renders with "Automatic selection" as default
4. Place another "Newsletter Ad" and click to Send
5. Confirm the preview renders both ads in the correct order (according to their position)
6. Close the modal and select the ad for each block (instead of automatic selection)
7. Confirm the choices are reflected when clicking to Send
8. Also confirm the newsletter's. "Ads Settings" shows the automatic insertion of ads disabled with proper notice, as in the image above

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-newsletters-ads.php [hunk 1]
```diff
@@ -202,12 +228,20 @@ public static function should_render_ads( $post_id ) {
...
 		 */
 		if ( get_post_meta( $post_id, 'disable_auto_ads', true ) ) {
 			$should_render_ads = false;
 		}
 
+		/**
+		 * Disable automated ads insertion if the newsletter contains a manual ad block.
+		 */
+		$content = get_the_content( null, false, $post_id );
+		if ( strpos( $content, '<!-- wp:newspack-newsletters/ad' ) !== false ) {
```
[the remark this answers is not in the document]
Comment (miguelpeixe) [reply includes/class-newspack-newsletters-ads.php]: Good call! Updated in ac452de6fdec705ae46af98056a1d660cfe0f6d2

**Review by dkoo — APPROVED** [review]
> That relates to our issue with the MJML strategy. There must be a save made through the editor to generate the final HTML markup because the MJML -> HTML transformation is in the browser.

I'm going to approve since this is a known shortcoming. Wonder if it's possible and worth re-processing the HTML upon clicking the "Send" or "Send test" buttons?

**Comment by miguelpeixe** [comment]
Hmm.. the test does save before sending. At which point are you modifying the ads before sending? Does the HTML preview does not match the sent body?

**Comment by dkoo** [comment]
> At which point are you modifying the ads before sending? Does the HTML preview does not match the sent body? Does the HTML preview does not match the sent body?

The HTML preview in the send modal (and in all actually sent, non-test emails) always matched the editor in my tests. It's just test emails sent via the Testing sidebar panel that sometimes don't match. The issue is replicable if you do the following:

1. Create a new newsletter
2. Insert an ad block
3. Without saving, send a test email

After manually saving the test email does always seem to match, so I think this is an edge case at most.

**Comment by miguelpeixe** [comment]
Ok! I'll merge this one but do some more testing and might have a patch for this before the epic merge. Thanks!
